src/mem/mod.rs

```diff
@@ -9,3 +9,25 @@ mod arena;
 pub use arena::*;
 
 pub mod cow;
+
+/// Aligns the given address to the alignment for the given type.
+///
+/// `align` must be a power of two; otherwise, the returned value
+/// will be well-defined but unspecified.
+///
+/// This function will always return a value greather than or equal to `addr`.
+#[inline]
+pub(in crate) fn align_to(addr: usize, align: usize) -> usize {
+    let mask = align.saturating_sub(1);
+    let (addr, overflow) = addr.overflowing_add(mask);
+    if overflow {
+        return usize::MAX;
+    }
+    addr & !mask
+}
+
+/// Computes the stride of `T`, that is, its size rounded up to its alignment.
+#[inline]
+pub(in crate) fn stride_of<T>() -> usize {
```

Comment:
`core::mem::size_of::<T>()` is guaranteed to be a multiple of `core::mem::align_of::<T>()` already. See the second sentence of the [size_of docs](https://doc.rust-lang.org/core/mem/fn.size_of.html):

> More specifically, this is the offset in bytes between successive elements in an array with that item type including alignment padding. Thus, for any type `T` and length `n`, `[T; n]` has a size of `n * size_of::<T>()`.

Rust's `size_of` refers to the same concept as Swift calls "stride". If Rust ever has a distinction akin to Swift's "size"/"stride" distinction, then Rust's equivalent to Swift's "size" will need to be called something else. Here's some more background on the topic: https://github.com/rust-lang/rust/issues/33266

Reply:
Wait really? I *swear* that Rust did not promise this. TIL!